models/biocro/R/met2model.BIOCRO.R

```diff
@@ -45,12 +44,34 @@ met2model.BIOCRO <- function(in.path, in.prefix, outfolder, overwrite = FALSE,
     }
 
     met.nc <- ncdf4::nc_open(ncfile)
-    tmp.met <- load.cfmet(met.nc, lat = lat, lon = lon,
-                          start.date = yrstart, end.date = yrend)
+    on.exit(ncdf4::nc_close(met.nc))
 
-    dt <- lubridate::as.period(mean(diff(tmp.met$date)))
-    if (dt > lubridate::hours(1)) {
-      tmp.met <- cfmet.downscale.time(cfmet = tmp.met, output.dt = 1)
+    dt <- mean(diff(udunits2::ud.convert(
+      met.nc$dim$time$vals,
+      met.nc$dim$time$units,
+      "hours since 1700-01-01 00:00:00")))
+    if (dt < 1) {
+      # More than one obs/hour. Write upscaled hourly file and reload.
+      on.exit()
+      ncdf4::nc_close(met.nc)
+
+      upscale_result <- PEcAn.data.atmosphere::upscale_met(
```

Comment:
I agree that this is an awkward place to do the scaling, especially since upscale writes a new CF file and downscale returns a dataframe, but is the desired temporal resolution available before this stage?

If keeping this, I should at least make sure to insert a Bety record for the upscaled file.